### PR TITLE
Issue #1686 Further clean ups

### PR DESCRIPTION
**Pull request by ljacomet**
* Refine interfaces
* clientID injection
* Split invalidation of hash and clear asks
* Remove store name from a bunch of messages

### CODE REVIEW

File: clustered/server/src/test/java/org/ehcache/clustered/server/ServerStoreCompatibilityTest.java [hunk 1]
```diff
@@ -51,18 +51,14 @@ public void testStoredKeyTypeMismatch() {
     ServerStoreConfiguration serverConfiguration = new ServerStoreConfiguration(DEDICATED_POOL_ALLOCATION,
                                                                                 STORED_KEY_TYPE,
                                                                                 STORED_VALUE_TYPE,
-                                                                                ACTUAL_KEY_TYPE,
-                                                                                ACTUAL_VALUE_TYPE,
-                                                                                KEY_SERIALIZER_TYPE,
+      KEY_SERIALIZER_TYPE,
```
Comment (albinsuresh) [inline clustered/server/src/test/java/org/ehcache/clustered/server/ServerStoreCompatibilityTest.java]: Again nitpicking on formatting.

**Comment by albinsuresh** [comment]
`SyncMessageType` is not relevant anymore. Can be gotten rid of. With that change `EhcacheSyncMessage` and `EhcacheDataSyncMessage` can be merged as well.

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/InternalEhcacheClientEntity.java [hunk 2]
```diff
@@ -16,9 +16,15 @@
...
+import java.util.UUID;
+
 /**
  * InternalEhcacheClientEntity
  */
 public interface InternalEhcacheClientEntity extends EhcacheClientEntity {
+
   void setTimeouts(Timeouts timeouts);
+
+  void setClientId(UUID clientId);
```
Comment (albinsuresh) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/InternalEhcacheClientEntity.java]: I do not see any usages of this method.

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java [hunk 3]
```diff
@@ -68,21 +70,21 @@ public void onResponse(EhcacheEntityResponse.ClientInvalidateHash response) {
         final long key = response.getKey();
         final int invalidationId = response.getInvalidationId();
 
-        if (cacheId.equals(messageFactory.getCacheId())) {
+        if (cacheId.equals(cacheId)) {
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java]: damn, right!

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java [hunk 4]
```diff
@@ -92,21 +94,21 @@ public void onResponse(EhcacheEntityResponse.ClientInvalidateAll response) {
         final String cacheId = response.getCacheId();
         final int invalidationId = response.getInvalidationId();
 
-        if (cacheId.equals(messageFactory.getCacheId())) {
+        if (cacheId.equals(cacheId)) {
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java]: yeah

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/ClusteredStoreTest.java [hunk 5]
```diff
@@ -94,14 +94,13 @@ public void setup() throws Exception {
     ServerStoreConfiguration serverStoreConfiguration =
         new ServerStoreConfiguration(resourcePool.getPoolAllocation(),
             Long.class.getName(), String.class.getName(),
-            Long.class.getName(), String.class.getName(),
-            LongSerializer.class.getName(), StringSerializer.class.getName(),
+          LongSerializer.class.getName(), StringSerializer.class.getName(),
```
Comment (ljacomet) [inline clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/ClusteredStoreTest.java]: Will clean up

File: clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ServerStoreOpCodec.java [hunk 6]
```diff
@@ -174,61 +176,59 @@ public EhcacheEntityMessage decode(EhcacheMessageType opCode, ByteBuffer message
...
-        String cacheId = decoder.string(SERVER_STORE_NAME_FIELD);
+        Long key = decoder.int64(KEY_FIELD);
         Integer invalidationId = decoder.int32("invalidationId");
-        ClientInvalidationAck message = new ClientInvalidationAck(cacheId, invalidationId);
+        ClientInvalidationAck message = new ClientInvalidationAck(key, invalidationId);
         message.setId(msgId);
         return message;
       }
+      case CLIENT_INVALIDATION_ALL_ACK: {
+        return new ClientInvalidationAllAck(CLIENT_INVALIDATION_ALL_ACK_MESSAGE_STRUCT.decoder(messageBuffer).int32("invalidationId"));
```
Comment (ljacomet) [inline clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ServerStoreOpCodec.java]: Fixing

File: clustered/server/src/test/java/org/ehcache/clustered/server/ServerStoreCompatibilityTest.java (hunk 1 above)
Comment (ljacomet) [inline clustered/server/src/test/java/org/ehcache/clustered/server/ServerStoreCompatibilityTest.java]: Will fix

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java (hunk 3 above)
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/CommonServerStoreProxy.java]: Made me realise that is yet another place where that `cacheId` in the message is useless.

**Comment by ljacomet** [comment]
@albinsuresh Actually still is relevant ... just that we completely forgot to test passive sync with custom serializers ... filing an issue to fix that.

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
Managed to sneak in an `ArrayIndexOutOfBounds` - latest push fixes it

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
Merging in the face of failures - to be investigated later.

**Pull request merged**